ip.go

```diff
@@ -0,0 +1,153 @@
+// Package ip filters HTTP traffic based on IP ranges.
```

Comment:
I thought about this. I would prefer keeping it isolated, but in the future we can depend on this package from `mux` for convenience if the use case is common enough.